Sources/AppcuesKit/Presentation/Actions/Appcues/AppcuesCloseAction.swift

```diff
@@ -17,14 +17,20 @@ internal class AppcuesCloseAction: AppcuesExperienceAction {
 
     static let type = "@appcues/close"
 
+    private weak var appcues: Appcues?
+
     private let markComplete: Bool
 
     required init?(configuration: AppcuesExperiencePluginConfiguration) {
+        appcues = configuration.appcues
```

Comment:
just thinking, alternative pattern would be to resolve the `experienceRender` right here and store a weak reference to that? then the execute would just be 
```
experienceRenderer?.dismissCurrentExperience(markComplete: markComplete) { _ in completion() }
```
any pros/cons either way that you can see? I suppose would still need some sort of guard on nil to ensure the completion was called. This seems fine, just thinking out loud. The `return completion()` syntax caught me off guard.

Reply:
Right. Both approaches are workable. I started off resolving the dependencies in the init but then I ended up doing them all with `Appcues` for the sake of consistency since some actions actually use the `Appcues` interface directly.

Not that it's a good reason, but referencing `Appcues` vs the dependencies also means fewer changes from the way it was before.

Reply:
yep, this all seems good to me as-is, thanks for indulging my ramblings